### PR TITLE
Scientific calculator

### PR DESCRIPTION
**Pull request by razarehan28**

### CODE REVIEW

File: calc.css [hunk 1]
```diff
@@ -0,0 +1,122 @@
+* {
```
Comment (desaianiruddh) [inline calc.css:1]: improve indentation for css

File: calc.css [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	margin: 0;
+	padding: 0;
+	box-sizing: border-box;
+}
+:root {
+	--container-bg-color: #f1f2f4;
```
Comment (desaianiruddh) [inline calc.css]: use rgb instead of hex code if other else places you've used rgb

File: calc.css [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	margin: 2rem auto;
+	width: 400px;
+}
+.calculator-screen {
+	height: 85px;
+	text-align: right;
+	width: 400px;
+	font-size: 40px;
+	border: none;
+	padding: 1rem;
```
Comment (desaianiruddh) [inline calc.css]: use px instead of rem

File: calc.css [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+	background-color: var(--screen-bg-color);
+	overflow: scroll hidden;
```
Comment (desaianiruddh) [inline calc.css]: use either scroll or hidden

File: calc.html [hunk 5]
```diff
@@ -0,0 +1,136 @@
+<!DOCTYPE html>
```
Comment (desaianiruddh) [inline calc.html:1]: improve indentation

File: calc.html [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+								<div class="btn" id="floor">Round</div>
+								<div class="btn" id="ceil">Absolute</div>
+							</div>
+						</div>
+
+					</div>
+				</div>
+				<!-- row-4 -->
+				<div class="row">
+					<div id="second" class="btn">2<sup>nd</sup></div>
```
Comment (desaianiruddh) [inline calc.html]: Here id name second is not specifying what is second. improve id and class name

File: calc.html [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+					<div id="absolute" class="btn">|x|</div>
+					<div id="exponent" class="btn">exp</div>
+					<div id="mod" class="btn">mod</div>
+				</div>
+				<!-- row-6 -->
+				<div class="row second">
+					<!-- unicode of square root -->
+					<div id="root" class="btn">&radic;x</div>
+					<!-- unicode of cube root -->
+					<div id="qube-root" class="btn hide">&#8731;x</div>
```
Comment (desaianiruddh) [inline calc.html]: id name should be cube not qube

File: calc.html [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+					<div id="open-parenthesis" class="btn">(</div>
```
Comment (desaianiruddh) [inline calc.html]: id name should be in camelCase Convention

File: calc.css [hunk 9]
```diff
@@ -0,0 +1,122 @@
+* {
+	margin: 0;
+	padding: 0;
+	box-sizing: border-box;
+}
+:root {
+	--container-bg-color: #f1f2f4;
+	--screen-bg-color: rgb(242, 242, 242);
+	--btn-hover-color: rgb(188, 226, 248);
+	--dropdownMenu-color: rgb(228, 240, 247);
```
Comment (Rutvi25) [inline calc.css]: use - instead of camel case for css variables.
Eg. dropdown-menu-color

File: calc.css [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+	--box-shadow-color: rgba(46, 46, 46, 1);
+	--equal-operator: #92c2e8;
+}
+.container {
+	background-color: var(--container-bg-color);
+	margin: 2rem auto;
```
Comment (Rutvi25) [inline calc.css]: either use rem or px, at every places

File: calc.css [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+	text-align: right;
+	width: 400px;
+	font-size: 40px;
+	border: none;
+	padding: 1rem;
+	background-color: var(--screen-bg-color);
+	overflow: scroll hidden;
+}
+body {
+	font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
```
Comment (Rutvi25) [inline calc.css]: unused fonts can be removed

File: calc.html [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+						<div id="memory-recall" class="btn">MR</div>
+						<div id="memory-plus" class="btn memory">M+</div>
+						<div id="memoy-minus" class="btn memory">M-</div>
+						<div id="memory-stored" class="btn memory">MS</div>
+					</div>
+					<!-- row-3 -->
+					<div class="row">
+						<div class="dropdown">
+							<button class="trigonometry-function dropdown-btn column-2 btn">
+								<img src="./Trigonometry.png" alt="trigo-icon" />
```
Comment (Rutvi25) [inline calc.html]: use trigonometry instead of trigo

File: calc.html [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+								Function
+							</button>
+							<div class="dropdown-chart-2">
+								<div class="btn" id="Absolute">Ceil</div>
+								<div class="btn" id="round">Floor</div>
+								<div class="btn" id="floor">Round</div>
+								<div class="btn" id="ceil">Absolute</div>
+							</div>
+						</div>
+
```
Comment (Rutvi25) [inline calc.html]: remove unnecessary line gaps

File: calc.js [hunk 14]
```diff
@@ -0,0 +1,516 @@
+
```
Comment (Rutvi25) [inline calc.js]: remove unnecessary line gaps

File: calc.js [hunk 15]
```diff
@@ -0,0 +1,516 @@
...
+const deg = document.querySelector("#degree");
+const functionButton = document.querySelector(".function-btn");
+const operator = document.querySelectorAll(".operator");
+const equal = document.querySelector("#equal");
+const trigoButton = document.querySelector(".trigonometry-function");
+const second = document.querySelector("#second");
+const number = document.querySelectorAll(".number");
+const screen = document.querySelector(".calculator-screen");
+const backspace = document.querySelector("#backspace");
+const del = document.querySelector("#clear");
```
Comment (Rutvi25) [inline calc.js]: use appropriate name, eg. delete

File: calc.js [hunk 16]
```diff
@@ -0,0 +1,516 @@
...
+//initialize the screen value
+let memory = [], i = 0;
+screen.innerHTML = "";
+let Flag = false;
+// 2nd button 
+second.addEventListener("click", function () {
+	if (!second.classList.contains("color")) {
```
Comment (Rutvi25) [inline calc.js]: improve indentation

File: calc.js [hunk 17]
```diff
@@ -0,0 +1,516 @@
...
+const trigoButton = document.querySelector(".trigonometry-function");
+const second = document.querySelector("#second");
+const number = document.querySelectorAll(".number");
+const screen = document.querySelector(".calculator-screen");
+const backspace = document.querySelector("#backspace");
+const del = document.querySelector("#clear");
+//initialize the screen value
+let memory = [], i = 0;
+screen.innerHTML = "";
+let Flag = false;
```
Comment (Rutvi25) [inline calc.js]: it should be flag instead of Flag

File: calc.js [hunk 18]
```diff
@@ -0,0 +1,516 @@
...
+			if (Flag) {
+				screen.innerHTML = Math.sec((screen.innerHTML * Math.PI) / 180).toPrecision(2);
+			} else {
+				screen.innerHTML = Math.sec(screen.innerHTML).toPrecision(10);
+			}
+			break;
+		case "ceil":
+			screen.innerHTML = Math.ceil(screen.innerHTML);
+			break;
+			break;
```
Comment (Rutvi25) [inline calc.js]: keep only one break

File: calc.js [hunk 19]
```diff
@@ -0,0 +1,516 @@
...
+		case "floor":
+			screen.innerHTML = Math.floor(screen.innerHTML);
+			break;
+		case "round":
```
Comment (Rutvi25) [inline calc.js]: break; statement is missing

File: calc.js [hunk 20]
```diff
@@ -0,0 +1,516 @@
...
+		case "backspace":
+			screen.innerHTML = backSpace(screen.innerHTML);
+			break;
+		case "clear":
+			//clear the screen 
+			screen.innerHTML = "";
+			break;
+		case "pi":
+			if (screen.innerHTML === " ") {
+				screen.innerHTML = 3.1428;
```
Comment (Rutvi25) [inline calc.js]: Math.PI can be used

File: calc.js [hunk 21]
```diff
@@ -0,0 +1,516 @@
...
+			if (screen.innerHTML === " ") {
+				alert("Enter the number you want");
+			}
+			else {
+				screen.innerHTML = changeSign(screen.innerHTML);
+			}
+			break;
+		case "exponential":
+			if (screen.innerHTML === "") {
+				screen.innerHTML = 2.7182;
```
Comment (Rutvi25) [inline calc.js]: Math.E can be used

**Review by Rutvi25 — CHANGES_REQUESTED** [review]

File: calc.css [hunk 22]
```diff
@@ -0,0 +1,123 @@
...
+  width: 140px;
+  position: absolute;
+  background-color: var(--dropdownMenu-color);
+}
+.dropdown-chart-2 {
+  display: none;
+  width: 115px;
+  position: absolute;
+  background-color: var(--dropdownMenu-color);
+}
```
Comment (ghost) [inline calc.css:73]: for this two classes we can combine the common property in one class and giving the different properties with new classes.

File: calc.js [hunk 23]
```diff
@@ -0,0 +1,502 @@
...
+      break;
+    case "mod":
+      screen.innerHTML = Math.abs(screen.innerHTML);
+      break;
+    case "base10":
+      screen.innerHTML = Math.pow(10, screen.innerHTML);
+      break;
+    case "logarithm":
+      if (screen.innerHTML === "") {
+        alert("Enter the number you want");
```
Comment (desaianiruddh) [inline calc.js]: make alert message appropriate. Don't use same alert message each place.. be like "First enter number for log"

File: calc.js [hunk 24]
```diff
@@ -0,0 +1,502 @@
...
+      }
+      break;
+    case "qube-root":
+      if (screen.innerHTML.includes("-")) {
+        screen.innerHTML = -1 * Math.pow(-1 * screen.innerHTML, 1 / 3);
+      } else {
+        screen.innerHTML = Math.pow(screen.innerHTML, 1 / 3);
+      }
+      break;
+    case "x-root-y":
```
Comment (desaianiruddh) [inline calc.js]: it may x raised to y not x root y

File: calc.js [hunk 25]
```diff
@@ -0,0 +1,502 @@
...
+function memoryPlus(num) {
+  if (memory.length === 0) {
+    screen.innerHTML = "";
+    memory.push(num);
+    console.log(memory);
+  } else {
+    memory.push(num);
+  }
+}
+function memoryMinus() {
```
Comment (desaianiruddh) [inline calc.js]: memory minus logic is not appropriate. this calculator function not remove number from memory. it save number in memory with negative value

**Review by desaianiruddh — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by Rutvi25 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: calc.html [hunk 26]
```diff
@@ -0,0 +1,133 @@
...
+              <div class="dropdown">
+                <button class="function-btn dropdown-btn column-3 btn">
+                  <img src="./function.png" alt="function-icon" />
+                  Function
+                </button>
+                <div class="dropdown-chart-2">
+                  <div class="btn" id="Absolute">Ceil</div>
+                  <div class="btn" id="round">Floor</div>
+                  <div class="btn" id="floor">Round</div>
+                  <div class="btn" id="ceil">Absolute</div>
```
Comment (sheth-jay) [inline calc.html]: ids are not matching with their values.e.g. for Ceil button id is absolute.

File: calc.js [hunk 27]
```diff
@@ -0,0 +1,471 @@
...
+const operator = document.querySelectorAll(".operator");
+const equal = document.querySelector("#equal");
+const trigoButton = document.querySelector(".trigonometry-function");
+const second = document.querySelector("#secondBtn");
+const number = document.querySelectorAll(".number");
+const screen = document.querySelector(".calculator-screen");
+const backspace = document.querySelector("#backspace");
+const clear = document.querySelector("#clear");
+//initialize the screen value
+  i = 0;
```
Comment (sheth-jay) [inline calc.js]: add let or const before declaring a variable.

**Review by sheth-jay — CHANGES_REQUESTED** [review]

**Review by sheth-jay — APPROVED** [review]

**Pull request merged**